andela/thor-ah
# 159987690 User should get verification email after successful signup

### What does this PR do?
Implement the feature that sends a verification email to a user after successful signup.
Confirms user's email after clicking on the verification link in the email.
Resend email verification to user when they request for it.

### Description of Task to be completed?
added two new endpoints to the routes
  POST /api/users/confirmation/:token
  POST /api/users/verify/resend-email
user should get a verification email after successful signup
user's email should be verified when they hit this route /api/users/confirmation/:token;
user can also request that the verification email be resent when they make a post request to this route /api/users/verify/resend-email

### How should this be manually tested?
After cloning the app
create a .env file from .envsample
run npm install
Install and setup Postgres
run npm start:dev to start the development server
on postman, make a POST request to /api/users to signup a new user
You should a message that a verification email has been sent to the user
Copy the link from the email to make a GET request to /api/users/confirmation/:token
You should get a message that the email has been confirmed
Request that verification email be resent by making a POST request to /api/users/verify/resend-email

### What are the relevant pivotal tracker stories?
[159987690](https://www.pivotaltracker.com/story/show/159987690)
